Add Travis CI

adamcaudill opened the pull request:
This adds Travis CI support.

Fixes #7, replaces #9.

Commit:
Add Travis config

adamcaudill commented:
As expected, failing due to the version check:

>  Invalid version of libeay32, expecting 1.0.2a Release, got: 1.0.1a Release

Need to update the `.travis.yml` file to add a step to download and build the proper OpenSSL version.

Commit:
Travis: Install OpenSSL from source

Commit:
Correct openssl archive name

Commit:
Better OpenSSL build script

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

flaub commented:
Any luck on this? Would be great to have CI coverage on windows, linux, and osx.

Commit:
Use OpenSSL ftp site, as web links aren't stable

It seems that for some reason the OpenSSL folks remove the file after a new release, so we have to use the FTP server to get a URL that will be valid if we are behind on the version we are using.

adamcaudill commented:
Latest update: As you can see from [the log](https://travis-ci.org/openssl-net/openssl-net/builds/67753065) it's still failing due to a version mismatch - running `openssl version` returns `OpenSSL 1.0.2a 19 Mar 2015` - so the install is certainly working. 

I started tracking this down back in March, but I don't recall now what was going on - my guess is that we're ending up with two versions of OpenSSL on the build server. Should be a simply fix, just need to reproduce and track down on my Ubuntu or Debian dev box.

flaub commented:
This issue was the primary reason I had originally distributed native libraries for all platforms. 

At least on my Ubuntu machine, `libcrypto.so.1.0.0` and `libssl.so.1.0.0` live under `/lib/x86_64-linux-gnu`.

We could try to place our custom build of OpenSSL libs in the same directory as the target, `bin/$CFG`.

flaub commented:
Interesting, do Travis linux machines come with `nuget` already installed?
